### ISSUE TITLE
CA1801 is reported in a event-handler method that taking DependencyPropertyChangedEventArgs

### ISSUE THREAD

**Issue by Tan90909090** [dotnet/roslyn-analyzers#2458]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2

#### Diagnostic ID

[CA1801](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801-review-unused-parameters)

#### Repro steps
Create a WPF project, write a following code:
```xml
<Window x:Class="WpfTest.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        IsEnabledChanged="Window_IsEnabledChanged" />
```
```cs
using System.Windows;

namespace WpfTest
{
    public partial class MainWindow : Window
    {
        public MainWindow() => InitializeComponent();

        // CA1801 is reported
        private void Window_IsEnabledChanged(object sender, DependencyPropertyChangedEventArgs e)
        {
        }
    }
}
```

#### Expected behavior
CA1801 is not reported. The method is an event handler so it must have parameters.

#### Actual behavior
CA1801 is reported:
`Parameter sender of method Window_IsEnabledChanged is never used. Remove the parameter or use it in the method body.`
`Parameter e of method Window_IsEnabledChanged is never used. Remove the parameter or use it in the method body.`

According to [a comment](https://github.com/dotnet/roslyn-analyzers/issues/2338#issuecomment-489806299), there is a case that the we handle the method as a special case:
> 1. 2 parameters
> 2. First parameter is of type object
> 3. Second parameter is of type EventArgs or is a sub-type of EventArgs

Let's also handle the method as a special case when the second parameter is of type [DependencyPropertyChangedEventArgs](https://docs.microsoft.com/en-us/dotnet/api/system.windows.dependencypropertychangedeventargs), which is not sub-type of EventArgs.

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/2593
